bitnami/contour/charts/contour-crds/Chart.yaml

```diff
@@ -0,0 +1,23 @@
+apiVersion: v1
+name: contour-crds
```

Comment:
Are you sure, that we need a subchart for this? As far as I understand about helm and CRDs, the real magic about  CRD management belongs to the `/crds/`-Directory in helm3 and the `crd-install`-hook in helm2.

When we do not apply resources with this contour-chart which belongs to the definitions we need to apply before, I don't see the usage of a subchart for the CRDs.

There are Pros and Cons for managing CRDs with helm using the `/crds/`-Directory / `crd-install`-hook. When the CRD is applied by helm using /crds/ or hooks, the CRD does not get deleted if we uninstall the release. Otherwise it gets deleted including all "instances" of this type (cascade delete) which could be dangerous.

Reply:
We could simply move the CRDs outside the `crds/` directory in the same chart, this does not work since we need to ensure CRDs are applied before everything.

Reply:
That is true for charts applying resources which belongs to the CRDs the chart introduces. The prometheus-operator chart is a good example for this.
But the contour chart currently do not apply a resource to the K8s-API which is an "instance" of the CRD it applies. The CRDs are used by other applications where the Ingress-v1-API does not provide enough features. Contour currently does not enter healthy/ready state when the CRDs are absent, but it get healthy at runtime as soon the CRDs exist. I maintain the CRDs outside of helm, so I have full control when new features arrive.

Reply:
I pulled out the CRDs into a subchart for a couple reasons:

1. Helm does not currently manage CRD lifecycle.
2. Subcharts will always run first. Even if contour does not instantiate CRs as part of its installation, I wanted to codify that order.

Granted there are risks in this approach as you mentioned - and I'm grateful that you raised them. I'll be happy to collaborate to try to get a solution we can all agree on.